### DISCUSSION TITLE
Add the tags default icon within (or next to) search bar leading to the tags index page

### DISCUSSION THREAD

**Discussion by Guts** [squidfunk/mkdocs-material#4667] [Ideas]
I'm pretty fan of the tags system and its impact on the search plugin but I think the tags index is quite unreachable for a documentation website using the theme.

It could be more visible and accessible through the search bar using the default tag icon. It deserves it, don't you think? :wink:

Clicking on this con could lead to the tags index page or, I'm a dreamer, switching the search preview to a list of clickable tags filtering the lunr search (with something like `+tags:{selected_tag} search term`).

**Comment by squidfunk** [comment]
Thanks, glad the functionality is useful to you! I'm not sure I understand your request though. Could you maybe create a quick mockup / screenshot that explains what you're aiming for?

**Comment by Guts** [comment]
Disclaimer: I'm really bad at playing with drawing tools and I'm still not really clear about how to implement my idea. do not forget my goal: make the tags more prominent in the global UI and more specifically in search.

Opening the search bar, add the tags icon on the right:

![search_with_tags_icon](https://user-images.githubusercontent.com/1596222/203629415-b3312076-d1f6-423b-9c74-c8aaf4b346c7.png)

Clicking on the icon: display list of available tags or a glossary view that allow end-user pick ones (with AND behavior) and filtering on search result automatically adding the adequated syntax. Example on your doc website, even if you don't have tags: https://squidfunk.github.io/mkdocs-material/blog/2021/09/26/excluding-content-from-search/?q=%2Bsearch+%2Btag

Example of glossary view that I've in mind for this and tags index:

![image](https://user-images.githubusercontent.com/1596222/203630408-f2b53f69-f233-439d-b441-90ba25d08796.png)

**Reply by squidfunk** [reply]
Ah I see. Thanks for suggesting. Nice idea, but let me say that tags are not the last thing to appear in search. In the mid-term, I want to allow authors to index structured data for each document, which means that more buttons would be needed. Let me think about this for a while – I'll take this idea with me when I start working on this.

**Reply by Guts** [reply]
> I want to allow authors to index structured data for each document,

What do you mean by structured data? What Google names rich snippets, based on http://schema.org and validated by https://validator.schema.org/?

FYI I'm already doing something like this through Jinja2 templating for Geotribu. Doc (in French, sorry): https://static.geotribu.fr/contribuer/internal/seo_extraits_enrichis/#exemples-de-donnees-structurees-generees-sur-geotribu

> Let me think about this for a while – I'll take this idea with me when I start working on this

Sure! If you need some help or an additional brain to think about, let me know.

**Reply by squidfunk** [reply]
Basically anything authors can define via front matter (YAML).
